proposals/kroxylicious-operator-api-alternative.md

```diff
@@ -0,0 +1,654 @@
+  # list of ingresses the virtual cluster wishes to use.  Each ingress selected must be one associated with
+  # the proxy.  It is legal for the virtualcluster to choose only a subset.
+  ingresses:
+  - name: myclusterip
+    tls:
+      # server certificate - grab bag matched by hostname (what about key/keystore password?)
+      certificateRefs:
+      - kind: Secret # if present must be Secret, otherwise defaulted to Secret
+        group: ""  # if present must be "", otherwise defaulted to ""
+        name: servercert
+        namespace: # namespace of the secret, if omitted assumes namespace of this resource
+     # peer trust
+     trustAnchorRefs:
+      - kind: ConfigMap # if present must be ConfigMap, otherwise defaulted to ConfigMap
+        group: ""  # if present must be "", otherwise defaulted to ""
+        name: servercert
+        namespace: # namespace of the configmap, if omitted assumes namespace of this resource
+      trustOptions:
+       clientAuth: REQUIRED
+    infrastructure:
+      # Controls that let the Developer influence the annotations/labels created on the ingress objects (Service, OpenShift Routes, TLSRoutes etc).
+      # If an annotation or label supplied here duplicates one provided by the ProxyIngress, the one defined by the ProxyIngress takes precendence. 
+      # The specific use case for this is https://docs.openshift.com/container-platform/4.17/security/certificates/service-serving-certificate.html#add-service-certificate_service-serving-certificate
+      # where the developer wishes their VC to use Certificates created by OpenShift (useful for ClusterIP type TLS).
+      annotations:
+         x: y
+      labels:
+         x: y
+
+  # Points to the cluster being proxied.  Can either be Strimzi Kafka resource or endpoint details.
+  targetCluster:
+    # one of: resourceRef or resource
+    resourceRef:
+      kind: Kafka  # must be Kafka
+      group: strimzi.io # must be strimzi.io
+      name: my-cluster
+      listenerName: listener # name of strimzi listener 
+    resource:
+       bootstrap: bootstrap:9092
+       protocol: TCP|TLS
+       nodeIdRanges:
+      - name: mybrokers
+        range:
+          startInclusive: 0
+          endExclusive: 3
+    tls:
+      # Optional - client auth
+      certificateRef:
+        kind: Secret # if present must be Secret, otherwise defaulted to Secret
+        group: ""  # if present must be "", otherwise defaulted to ""
+        name: servercert
+        namespace: # namespace of the secret, if omitted assumes namespace of this resource
+      # Optional - peer trust
+      trustAnchorRefs:
+       - kind: ConfigMap # if present must be ConfigMap, otherwise defaulted to ConfigMap
+         group: ""  # if present must be "", otherwise defaulted to ""
+         name: servercert
+         namespace: # namespace of the configmap, if omitted assumes namespace of this resource
+
+    # ordered list of filters to be used by the virtualcluster
+    filterRefs:
```

Comment:
I could have screwed up but I see the filterRefs as being scoped to the target cluster and I think they should be scoped to the virtual cluster. 

I'd also argue that the property should be `filterChain` to make the ordering concerns clear and obvious.

Reply:
> I could have screwed up but I see the filterRefs as being scoped to the target cluster and I think they should be scoped to the virtual cluster.

> I'd also argue that the property should be filterChain to make the ordering concerns clear and obvious.

I've used `*Refs` across the API, borrowing the terminology from `GatewayAPI`.   So that gives me `filterRefs`.   `filterChainRefs` wouldn't really make sense.  You'd expect to be listing FilterChain CRs.

Reply:
Hmm consistency vs domain specific naming. Tough one. I can certainly live with `filterRefs`, though maybe we should think about a `filterChain` CR as well, but I think `filterRefs` is the right level for MVP.